### ISSUE TITLE
Incorrect: Unnecessary semicolon ERror

### ISSUE THREAD

**Issue by mhevery** [palantir/tslint#2162]
### Bug Report

- __TSLint version__: `4.4.2`
- __TypeScript version__: `2.0`
- __Running TSLint via__: (please choose one) grunt-tslint 

#### TypeScript code being linted

```ts
class FormControlSelectWithCompareFn {
  compareFn: (o1: any, o2: any) => boolean = (o1: any, o2: any) => {
    return o1 && o2 ? o1.id === o2.id : o1 === o2;
  };  <<<< PROBLEM HERE
  cities = [{id: 1, name: 'SF'}, {id: 2, name: 'NY'}];
  form = new FormGroup({city: new FormControl({id: 1, name: 'SF'})});
}
```

with `tslint.json` configuration:

```json
{
  "rulesDirectory": [
    "dist/tools/tslint",
    "node_modules/vrsource-tslint-rules/rules",
    "node_modules/tslint-eslint-rules/dist/rules"
  ],
  "rules": {
    "file-header": [true, "Copyright Google Inc\\."],
    "no-console": [true, "log"],
    "no-duplicate-imports": true,
    "no-duplicate-variable": true,
    "no-jasmine-focus": true,
    "no-var-keyword": true,
    "require-internal-with-underscore": true,
    "semicolon": [true],
    "variable-name": [true, "ban-keywords"],
    "no-inner-declarations": [true, "function"]
  }
}
```

#### Actual behavior

`Unnecessary semicolon`

#### Expected behavior

pass with no errors

**Comment by adidahiya** [comment]
duplicate of #2015
